docs/csharp/language-reference/compiler-messages/partial-declarations.md

```diff
@@ -0,0 +1,231 @@
+- **CS0764**: *Both partial method declarations must be `unsafe` or neither may be `unsafe`*
+- **CS8142**: *Both partial member declarations must use the same tuple element names.*
+- **CS8663**: *Both partial member declarations must be readonly or neither may be readonly*
+- **CS8799**: *Both partial member declarations must have identical accessibility modifiers.*
+- **CS8800**: *Both partial member declarations must have identical combinations of `virtual`, `override`, `sealed`, and `new` modifiers.*
+- **CS8818**: *Partial member declarations must have matching `ref` return values.*
+- **CS8988**: *The `scoped` modifier of parameter doesn't match partial definition.*
+
+Partial members have two declarations. The declaration without an implementation is the *declaring declaration*. The declaration with the implementation is the *implementing declaration*. Partial members are allowed only in a `partial` type. Partial members can't be `abstract`. Partial members can't explicitly implement an interface. Both declarations of a partial member must have identical signatures. For example, either both or neither declarations can include the `static` or `unsafe` modifiers.
+
+## Partial methods
+
+- **CS0501**: *'member function' must declare a body because it is not marked `abstract`, `extern`, or `partial`*
+- **CS0755**: *Both partial method declarations must be extension methods or neither may be an extension method.*
+- **CS0756**: *A partial method may not have multiple defining declarations.*
+- **CS0757**: *A partial method may not have multiple implementing declarations.*
+- **CS0759**: *No defining declaration found for implementing declaration of partial method.*
+- **CS0761**: *Partial method declarations of `method<T>` have inconsistent type parameter constraints.*
+- **CS0762**: *Cannot create delegate from method because it is a partial method without an implementing declaration*
+- **CS1067**: *Partial declarations must have the same type parameter names and variance modifiers in the same order.*
+- **CS8796**: *Partial method must have accessibility modifiers because it has a non-void return type.*
+- **CS8795**: *Partial member must have an implementation part because it has accessibility modifiers.*
+- **CS8797**: *Partial method  must have accessibility modifiers because it has '`out`' parameters.*
+- **CS8798**: *Partial method must have accessibility modifiers because it has a '`virtual`', '`override`', '`sealed`', '`new`', or '`extern`' modifier.*
+- **CS8817**: *Both partial method declarations must have the same return type.*
+
+Certain `partial` method declarations don't require an *implementing declaration*. If the member returns `void`, doesn't declare any access modifiers (including the default `private` modifier), and doesn't include any of the `virtual`, `override`, `sealed`, or `new` modifiers. Otherwise, any partial method must include both the declaring and implementing declarations.
```

Comment:
```suggestion
Certain `partial` method declarations don't require an *implementing declaration*. That is, if the member returns `void`, doesn't declare any access modifiers (including the default `private` modifier), and doesn't include any of the `virtual`, `override`, `sealed`, or `new` modifiers. Otherwise, any partial method must include both the declaring and implementing declarations.
```